torch/testing/_internal/common_nn.py

```diff
@@ -4867,15 +4867,30 @@ def check_jacobian(self, module, input: _TensorOrTensors, jacobian_input=True):
             self.assertLessEqual(max(differences), PRECISION)  # type: ignore[type-var]
 
 
-class TestBase(object):
+def is_criterion_test(test_params):
+    """
+    Helper function to identify whether or not the test specified by the given test_params is a
+    criterion test.
+
+    Args:
+        test_params (dict): Parameters defining the test
+
+    Returns:
+        True if the given params indicate a criterion test; False otherwise
+    """
+    return any([name in test_params for name in ['target', 'target_fn', 'target_size']])
 
-    _required_arg_names = {'constructor_args', 'input', 'extra_args'}
 
+class ModuleTest(object):
```

Comment:
Add a comment explaining this class and what it's intended to be used for + comment for each argument about how it's used